robocop/checkers/naming.py

```diff
@@ -227,3 +229,44 @@ def visit_Documentation(self, node):  # noqa
     def check_setting_name(self, name, node):
         if not (name.istitle() or name.isupper()):
             self.report("setting-name-not-capitalized", node=node)
+
+
+class VariableNamingChecker(VisitorChecker):
+    rules = {
+        "0309": (
+            "section-variable-not-uppercase",
+            "Section variable name should be uppercase",
+            RuleSeverity.WARNING
+        ),
+        "0310": (
+            "non-local-variables-should-be-uppercase",
+            "Test, suite and global variables should be uppercased",
+            RuleSeverity.WARNING
+        )
+    }
+
+    def __init__(self, *args):
+        self.set_variable_variants = {'settaskvariable',
+                                      'settestvariable',
+                                      'setsuitevariable',
+                                      'setglobalvariable'}
+        super().__init__(*args)
+
+    def visit_VariableSection(self, node):  # noqa
+        for child in node.body:
+            if not child.data_tokens:
+                continue
+            token = child.data_tokens[0]
+            if token.type == Token.VARIABLE and not token.value.isupper():
+                self.report("section-variable-not-uppercase", lineno=token.lineno,
+                            col=token.col_offset)
+
+    def visit_KeywordCall(self, node):  # noqa
+        if not node.keyword:
+            return
+        if normalize_robot_name(node.keyword) in self.set_variable_variants:
+            if len(node.data_tokens) < 2:
+                return
+            token = node.data_tokens[1]
+            if token.type == Token.ARGUMENT and not token.value.isupper():
```

Comment:
Are numbers in variable names ignored? Ie ${VERSION2}

Reply:
Yes, they are. Is it OK?

Reply:
Yeah that's good then